nntrainer/layers/conv2d_layer.h

```diff
@@ -121,6 +121,29 @@ class Conv2DLayer : public Layer {
    */
   void scaleSize(float scalesize) noexcept;
 
+  /**
+   * @brief     reform the data to 2d matrix
+   * a region is sampled considering @a padding, @a mstride of unit @a kdim
+   * Each region is mapped to one column,
+   * if channel mode, kernel channel is considered part of kernel feature
+   * if not, kernel channel is consider part of output dimension
+   *
+   * @param[in] in input data
+   * @param[in] kdim kernel dimesion for define number of row
+   * @param[in] padding padding information
+   * @param[in] mstride stride value : x, y direction
+   * @param[in] dilation kernel dilation factor : x, y each
+   * @param[in] channel_mode loop with channel first
+   * @param[out] out out tensor to put, if uninitialized, allocate a new tensor
+   * and set padding
+   * @note if out is initialized tensor, setting padding is skipped.
+   */
+  static void im2col(const Tensor &in, const TensorDim &kdim,
```

Comment:
I wonder why this is public?

Reply:
I intentionally aligned this to https://github.com/nnstreamer/nntrainer/pull/922/files#diff-96ba836ffd47c0f12c47c69ec6a55741d43c093ca1c8b35239b3039f857ab080R140-R143

Is the PR okay when I hide this function? 
If that's the case, the function doesn't need to be exposed in the header.

Reply:
That PR will need more changes than just im2col being public. For now, let's keep it private or outside the header.

Reply:
Done :)